cmake/CompileProtos.cmake

```diff
@@ -231,6 +231,8 @@ endfunction ()
 #
 # ${EXTERNAL_GOOGLEAPIS_SOURCE}/google/bigtable/v2/bigtable.proto
 #
+# ARGV2 can be specified as an optional path to substitute in lieu of
```

Comment:
The convention in CMake is to use `cmake_parse_arguments()`:

https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html

You can define a single-argument optional, e.g.

```cmake
function (google_cloud_cpp_load_protolist var file)
  cmake_parse_arguments(_opt
   # No boolean flags
   "" 
   # If present PROTO_DIR overrides the default "${EXTERNAL_GOOGLEAPIS_SOURCE}"
   "PROTO_DIR"
   # No multi-argument flags
    ""
  ${ARGN})
 if (NOT DEFINED _opt_PROTO_DIR)
   set(_opt_PROTO_DIR "${EXTERNAL_GOOGLEAPIS_SOURCE}")
 endif ()
...

```